### PR TITLE
Calzim/only example cleanup

### PR DESCRIPTION
**Pull request by CalZim**

### CODE REVIEW

File: basic/02c_feedback_loading_log.py [hunk 1]
```diff
@@ -5,9 +5,13 @@
...
 name = 'exampleLog.hebilog'
 
-log = hebi.util.load_log('{}/{}'.format(logFileFolder, logFileName))
+log = hebi.util.load_log('{}/{}'.format(folder, name))
 
 # Plot using some handy helper functions
 hebi.util.plot_logs(log, 'position', figure_spec=101)
+
+log = hebi.util.load_log('{}/{}'.format(folder, name))
 hebi.util.plot_logs(log, 'velocity', figure_spec=102)
```
Comment (rLinks234) [inline basic/02c_feedback_loading_log.py]: You can call `plot_logs` with a string object too, so something like:

```py
hebi.util.plot_logs('{}/{}'.format(folder, name), 'velocity', figure_spec=102)
```

but this is sufficient for now.

File: basic/03a_command_position.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 freq    = freq_hz * 2.0 * pi  # [rad / sec]
 amp     = pi * 0.25           # [rad] (45 degrees)
 
-duration = 10.0               # [sec]
+duration = 4              # [sec]
```
Comment (rLinks234) [inline basic/03a_command_position.py]: @iamtesch @CalZim Does anyone know why this changed to 4 seconds? It appears to be 10 seconds in the [matlab example](https://github.com/HebiRobotics/hebi-matlab-examples/blob/master/basic/x-series_actuator/ex3a_command_position.m#L34)

File: basic/03b_command_velocity.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
 freq    = freq_hz * 2.0 * pi  # [rad / sec]
 amp     = 1.0                 # [rad] (45 degrees)
 
-duration = 10.0               # [sec]
+duration = 4              # [sec]
```
Comment (rLinks234) [inline basic/03b_command_velocity.py]: To avoid spamming this PR with duplicate comments, see previous comment corresponding to `10->4` seconds in the examples... Not a big deal, just wondering why it was done

File: basic/03d_command_pos_vel.py [hunk 4]
```diff
@@ -47,5 +48,30 @@
 
 # Stop logging. `log_file` contains the contents of the file
 log_file = group.stop_log()
-hebi.util.plot_logs(log_file, 'position', figure_spec=101)
```
Comment (rLinks234) [inline basic/03d_command_pos_vel.py]: There was a bug in the Python API, which I _believe_ has since been fixed, related to this (first `plot_logs` would work, but second would not, since the log file had been iterated through)... If it's fixed, we can revert the changes on the RHS here back to what it was previously.

File: basic/07b_robot_6_dof_arm.py [hunk 5]
```diff
@@ -109,7 +111,13 @@ def execute_trajectory(group, model, trajectory, feedback):
 group.get_next_feedback(reuse_fbk=feedback)
 waypoints[:, 0] = feedback.position
 waypoints[:, 1] = joint_targets[:, 0]
+print("joint targets")
```
Comment (rLinks234) [inline basic/07b_robot_6_dof_arm.py]: These print statements are extraneous and should be removed for the public facing examples.

File: basic/mobile_io.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+
+class MobileIO():
+    
+    num_buttons = 8
+    num_sliders = 8
+    current_button_state = [0, 0, 0, 0, 0, 0, 0, 0]
+    current_slider_state = [0, 0, 0, 0, 0, 0, 0, 0]
+    
+    btn_states = ["", "", "", "", "", "", "", ""]
```
Comment (rLinks234) [inline basic/mobile_io.py]: These variables are class fields, not _instance_ fields (_i.e._, they are static variables). This means that multiple instances of `MobileIO` will refer to a single instance of these variables, which I assume is not the intention here.

Move these definitions into `__init__` to make them instance variables

File: basic/mobile_io.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        for i in range(self.num_sliders):
+            if self.fbk.io.a.has_int(i+1):
+                self.current_slider_state[i] = int(self.fbk.io.a.get_int(i+1))
+            elif self.fbk.io.a.has_float(i+1):
+                self.current_slider_state[i] = float(self.fbk.io.a.get_float(i+1))
+        
+        return (self.current_button_state, self.current_slider_state)
+    
+    # returns the change in button positions (currently breaks if called in same step as getState())
+    def getDiff(self):
```
Comment (rLinks234) [inline basic/mobile_io.py]: Generally, we have `snake_case` naming convention for our Python stuff (that is what is generally used in Python)... I'm not really opposed to `camelCase` (I prefer camel tbh), but it goes against what we generally use.

File: basic/mobile_io.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    
+    # clear text (currently does not work)
+    def clearText(self): 
+        self.cmd.clear_log = True # bugged needs to be looked at
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+    
+    # send a buzz to the mobile device (only works on iphones)
+    def sendVibrate(self):
+        self.cmd.effort = 1
+        return self.grp.send_command_with_acknowledgement(self.cmd)
```
Comment (rLinks234) [inline basic/mobile_io.py]: @iamtesch , should we do `send_command_with_ack` here, or just `send_command` ?

File: basic/mobile_io/2a_feedback.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+from time import sleep, time
+from matplotlib import pyplot as plt
+
+lookup = hebi.Lookup()
+
+# Wait 2 seconds for the module list to populate
+sleep(2.0)
+
+family_name = "HEBI" 
+module_name = "Cal's iPhone"
```
Comment (rLinks234) [inline basic/mobile_io/2a_feedback.py]: Make sure to change this back to what other examples are using (for consistency)

File: basic/mobile_io/2c_feedback_orientation.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+end_time = start_time + duration
+current_time = start_time
+
+while current_time < end_time:
+  current_time = time()
+  fbk = group.get_next_feedback()
+  orient = fbk[0].ar_orientation
+  
+  r = R.from_quat(orient)
+  rot_mat = r.as_matrix()
```
Comment (rLinks234) [inline basic/mobile_io/2c_feedback_orientation.py]: We have a function (`util.math_utils.quat2rot`) to do this, and I'd prefer not to bring in a dependency like `scipy` (I forget if it's a different package than `numpy` or what...). However... it's not imported here (you need to add a directory to the `PATH` for Python to find it), so this warrants a discussion on the best way to resolve this dependency.

File: basic/mobile_io/2d_feedback_full_pose.py [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  ])
+  
+  z = np.array([
+      [0, 0],
+      [0, 0],
+      [0, 1]
+  ])
+  
+  plt.clf()
+  ax = plt.axes(projection="3d")
```
Comment (rLinks234) [inline basic/mobile_io/2d_feedback_full_pose.py]: I'd assume this is potentially an expensive call which should be done once (outside of the loop), as opposed to each iteration

File: kits/arm/ex_mobile_device_control.py [hunk 12]
```diff
@@ -98,8 +93,18 @@ def get_ik(xyz_target, rot_target, ik_seed):
...
+counter = 0
 # Startup
 while not abort_flag:
+  # Update mobile io state
+  state = m.getState()
+  print("counter: " + str(counter))
+  print("state: " + str(state))
+  counter += 1
+  if counter % 100 == 0:
+      print("counter: " + str(counter))
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control.py]: @iamtesch , should we leave all/some of these print statements in?

File: kits/arm/ex_mobile_device_control.py [hunk 13]
```diff
@@ -111,106 +116,117 @@ def get_ik(xyz_target, rot_target, ik_seed):
   # Start background logging
   if enable_logging:
     group.start_log(os.path.join(local_dir, 'logs'), mkdirs=True)
-    phone_group.start_log(os.path.join(local_dir, 'logs'), mkdirs=True)
+    #phone_group.start_log(os.path.join(local_dir, 'logs'), mkdirs=True)
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control.py]: Do we not want to log mobile IO feedback? Just wondering. If we don't (as implied by this line being commented out), then we should delete this commented out line.

File: kits/arm/ex_mobile_device_control.py [hunk 14]
```diff
@@ -111,106 +116,117 @@ def get_ik(xyz_target, rot_target, ik_seed):
...
+  phone_fbk_timer = perf_counter()
 
   time_last = t0
   arm_trajStartTime = t0
 
-  first_run = true
-
+  first_run = True
+  
+  print("check 2")
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control.py]: I'm assuming these `check X` print statements were for debugging purposes... We should remove them before merging this in

File: kits/arm/ex_mobile_device_control.py [hunk 15]
```diff
@@ -224,14 +240,21 @@ def get_ik(xyz_target, rot_target, ik_seed):
...
+      a = [0, arm_traj_min_duration]
+      b = np.array([pos, ik_pos]).T
+      c = np.array([vel, end_velocities]).T
+      d = np.array([accel, end_accels]).T
+      arm_traj = create_trajectory(a, b, c, d)  
+    
+    print("check 4")
     # Send to robot
-    group.send_command(cmd)
+    #group.send_command(cmd)
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control.py]: This looks like it should be uncommented, right?

File: kits/arm/ex_mobile_device_control_v2.py [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+import numpy as np
+import os
+import sys
+
+
+# ------------------------------------------------------------------------------
+# Add the root folder of the repository to the search path for modules
+root_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+sys.path = [root_path] + sys.path
+# ------------------------------------------------------------------------------
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control_v2.py]: This is the hack necessary to add in the `util` package for stuff like the `quat2rot` functions.

File: kits/arm/ex_mobile_device_control_v2.py [hunk 17]
```diff
@@ -0,0 +1,294 @@
...
+root_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path = [root_path] + sys.path
+from util import math_utils
+
+enable_logging = False
+enable_effort_comp = True
+
+# Mobile Device Setup
+phone_family = 'HEBI'
+phone_name = "Cal's iPhone"
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control_v2.py]: Remember to change this back

File: kits/arm/ex_mobile_device_control_v4.py [hunk 18]
```diff
@@ -0,0 +1,320 @@
+import hebi
```
Comment (rLinks234) [inline kits/arm/ex_mobile_device_control_v4.py]: What is up with the `ex_mobile_device_control_vX.py` scheme here? It sounds like these were iterations to get the example working... If this is the case, we need to delete all but the one we want to keep

**Comment by rLinks234** [comment]
So far, these changes are pertinent to get this merged in:

- [ ] Remove extraneous print statements

- [ ] Be consistent with the `family`, `name` values (_e.g._ `Cal's phone` --> `Mobile IO` or whatever)

- [ ] Remove 3 of the 4 "iterations" of the mobile device example 

- [ ] Remove the `hebilog` files (These should not be in the repo). We can make sure to avoid any future `.hebilog` files creeping in to commits by adding a rule to the `.gitignore`

- [ ] `mobile_io.py` file is in the examples. This should live somewhere else. Probably under `util/input`

File: kits/arm/ex_mobile_device_control_v3.py [hunk 19]
```diff
@@ -0,0 +1,311 @@
...
+
+run_mode = "startup"
+
+enable_logging = True
+enable_effort_comp = True
+enable_ar_smoothing = True
+
+# Mobile device setup
+phone_family = 'HEBI'
+phone_name = "Cal's iPhone"
```
Comment (cwbollinger) [inline kits/arm/ex_mobile_device_control_v3.py]: fixme

File: kits/arm/mobile_io_control.py [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+import mobile_io as mbio
+
+
+run_mode = "points"
+
+first_run = True
+
+# Mobile device setup
+phone_family = 'HEBI'
+phone_name = "Phone Name"
```
Comment (cwbollinger) [inline kits/arm/mobile_io_control.py]: mobileIO

File: kits/arm/teach_repeat.py [hunk 21]
```diff
@@ -188,36 +189,42 @@ def run():
...
-
-  res = getch()
-
-  while res != 'q':
-    print_and_cr('')
+  print_and_cr("Press 'b2' to add waypoint ('b3' for stopping at this waypoint), 'b4' to clear waypoints, 'b5' to playback, and 'b1' to quit.")
+  print_and_cr("When in playback mode, 'b6' resumes training, and 'b1' quits.")
+  
+  
+  m = mbio.MobileIO("HEBI", "Phone Name")
```
Comment (cwbollinger) [inline kits/arm/teach_repeat.py]: mobileIO

**Comment by cwbollinger** [comment]
Also, do we want to have logfiles included in the examples repo? It doesn't bother me either way, but I'm not sure what they represent since their names are timestamps. Maybe rename them if they are for specific examples?

File: basic/03a_command_position.py (hunk 2 above)
Comment (iamtesch) [reply basic/03a_command_position.py]: Not sure -- I think this might have just been a testing change.  I'd prefer to keep it matching the MATLAB example for consistency.

File: basic/03d_command_pos_vel.py (hunk 4 above)
Comment (iamtesch) [reply basic/03d_command_pos_vel.py]: :+1: this would be great to test and revert if possible :)

File: basic/mobile_io.py (hunk 7 above)
Comment (iamtesch) [reply basic/mobile_io.py]: Yeah, as much as I dislike `snake_case`, I agree we should keep things consistent.  I have a feeling this current name was influenced by porting this from C++.

File: basic/mobile_io.py (hunk 8 above)
Comment (iamtesch) [reply basic/mobile_io.py]: Hmmm...I would probably stick with `send_command`, unless we are actually doing something with the feedback.

@CalZim , the biggest difference here is that under the covers, `send_command_with_ack` sends a message and then waits for a response.  This means that this call is blocking, and if the return packet is dropped will block until the timeout is reached.

`sendVibrate` is something that is more likely to happen in a control loop as feedback, so the "fire and forget" style of `send_command` is probably more appropriate.

For functions such as `setSnap` which configure the interface as a "one time" action, these are much more appropriate to be done with a `send_command_with_acknowledgement` and have wrapping code in the app to check for success.

File: basic/mobile_io/2c_feedback_orientation.py (hunk 10 above)
Comment (iamtesch) [reply basic/mobile_io/2c_feedback_orientation.py]: @CalZim -- I agree we should use our existing library functions where possible.
@rLinks234 -- I assume you mean the import is for the scipy/numpy, or is that for ours?

File: kits/arm/ex_mobile_device_control.py (hunk 12 above)
Comment (iamtesch) [reply kits/arm/ex_mobile_device_control.py]: I don't think so -- I think the `counter` was part of a debugging process? We should clean up the temporary/debug code where possible

File: kits/arm/ex_mobile_device_control.py [hunk 22]
```diff
@@ -111,106 +116,117 @@ def get_ik(xyz_target, rot_target, ik_seed):
...
   ik_pos = get_ik(xyz_target_init, rot_mat_target_init, ik_seed_pos)
 
   arm_traj = create_trajectory([0, arm_traj_min_duration], np.array([fbk.position, ik_pos]).T)
   fbk_time = fbk.receive_time
   t0 = fbk_time.min()
+  t_start = time()
   t = 0
-
+  #t = 0
+  print("check 1")
```
Comment (iamtesch) [inline kits/arm/ex_mobile_device_control.py]: More debug printout and commented variables here...

File: kits/arm/ex_mobile_device_control.py (hunk 12 above)
Comment (iamtesch) [reply kits/arm/ex_mobile_device_control.py]: I don't think so...

**Comment by iamtesch** [comment]
> Also, do we want to have logfiles included in the examples repo? It doesn't bother me either way, but I'm not sure what they represent since their names are timestamps. Maybe rename them if they are for specific examples?

Unless the logfiles are part of an example in loading logfiles, we shouldn't have these in general
